test/util/range.cpp

```diff
@@ -213,6 +213,21 @@ void test_pointers()
     BOOST_CHECK(bgr::at(r2, 5) == 5);
 }
 
+void test_const_pointers()
```

Comment:
That's indeed a more elegant solution. I'll amend the commit with your implementation.

BTW, I have an other PR in preparation that reworks the tests ([util] tests for now). The big picture is: 
- In Jamfile.v2 use `unit-test` rather than the deprecate `test-suite` rule
- Use more functionality of Boost.Test (such as automated test suite, templated test cases) 
- Use the prebuild Boost.Test (not the inline one) framework.

The benefits are:
- A better test report when invoking  `b2 test`
- The possibility to run a single test suite (e.g. `b2 range`) .
- Tests are faster to compile